src/clients/UBAClient/UBAClientTypes.ts

```diff
@@ -0,0 +1,65 @@
+import { BigNumber } from "ethers";
+import { UbaFlow } from "../../interfaces";
+import { UBAActionType } from "../../UBAFeeCalculator/UBAFeeTypes";
+import UBAConfig, { FlowTupleParameters } from "../../UBAFeeCalculator/UBAFeeConfig";
+
+export type RequestValidReturnType = { valid: false; reason: string } | { valid: true };
+export type OpeningBalanceReturnType = { blockNumber: number; spokePoolBalance: BigNumber };
+export type BalancingFeeReturnType = { balancingFee: BigNumber; actionType: UBAActionType };
+export type SystemFeeResult = { lpFee: BigNumber; depositBalancingFee: BigNumber; systemFee: BigNumber };
+export type RelayerFeeResult = {
+  relayerGasFee: BigNumber;
+  relayerCapitalFee: BigNumber;
+  relayerBalancingFee: BigNumber;
+  relayerFee: BigNumber;
+  amountTooLow: boolean;
+};
+
+export type UBAChainState = {
+  spokeChain: {
+    deploymentBlockNumber: number;
+    bundleEndBlockNumber: number;
+    latestBlockNumber: number;
+  };
+  bundles: {
+    [tokenSymbol: string]: UBABundleState[];
+  };
+};
+
+export type UBABundleState = {
+  openingBalance: BigNumber;
+  openingIncentiveBalance: BigNumber;
+  blockNumber: number;
```

Comment:
Should this be renamed `bundleEndBlockNumber`? https://github.com/across-protocol/sdk-v2/pull/266/files#r1247925856

Reply:
This is more apt as `openingBlockNumberForSpokeChain` because it denotes the `start` of the current bundle (since the last verified block bundle): https://github.com/across-protocol/sdk-v2/pull/266/commits/5334cac2b391dc7f07f65c48c186bb67fe1fcdc9